src/cargo/util/progress.rs

```diff
@@ -280,6 +353,7 @@ impl<'gctx> State<'gctx> {
         // return back to the beginning of the line for the next print.
         self.try_update_max_width();
         if let Some(pbar) = self.format.progress(cur, max) {
+            self.taskbar.percent = ((cur as f32) / (max as f32) * 100.0) as u8;
```

Comment:
`Format` has a `progress` function to abstract things like this.  I feel like `self.taskbar` should as well.

Reply:
I've reworked code a little: moved `TaskbarProgress` into `Format`. That also solved the problem with this line. No code duplication now, I've reused a value calculated and normalized in `Format::progress()`.

Now `Format::progress()` also returns not just a string, but a `ProgressOutput` enum with 3 variants: `PrintNow` (used in `Progress::print_now()`), `TextAndTaskbar` and `Taskbar` (for cases when we don't have space or terminal width is low).

I believe this makes intents clear, otherwise code in `Context::print()` becames too complicated because of all those corner cases when we want to output taskbar progress but not text.